xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-ui/src/main/resources/CKEditor/FileUploader.xml

```diff
@@ -103,12 +103,17 @@
       #set ($document = $xwiki.getDocument($request.document))
     #end
     #set ($reference = $document.documentReference)
-    #set ($attachment = $services.temporaryAttachments.uploadTemporaryAttachment($reference, 'upload', 
-        $request.filename))
-    #if ($attachment)
-      #sendSuccess($document, $attachment.filename)
+    #set ($skipRightCheck = ($request.getHeader('X-XWiki-Temporary-Attachment-Skip-Right-Check') == 'true'))
```

Comment:
Allowing file uploads without edit rights doesn't seem like a good idea as file uploads could easily lead to DoS attacks by filling the storage (which may be small in the case of cloud instances). I understand you probably need this for change requests but couldn't there at least be a configuration setting to allow this override? Or maybe add some listener that allows change request to explicitly authorize file uploads if the current user is allowed to create a change request for the current document instead of just always allowing file uploads?